src/software/estop/UARTCommunication.cpp

```diff
@@ -0,0 +1,67 @@
+
+#include "UARTCommunication.h"
+
+#include <boost/asio.hpp>
+#include <g3log/loglevels.hpp>
+#include <g3log/g3log.hpp>
+#include <iostream>
+#include <boost/function.hpp>
+#include <boost/exception/all.hpp>
+
+UARTCommunication::UARTCommunication(){}
+
+
+int UARTCommunication::openPort(io_service &ioService, int baudRate, std::string port)
```

Comment:
I'm curious if the parameter `port` being of type `int` and then converted to a `string` would make more sense for the caller. Instead of the caller passing the port number as a string?

Reply:
in this context `port` refers to the device that we will be communicating with, and is given as a string like `"/dev/ttyACM0"`. Will change the name to `serial_port_device` to better reflect this.